lib/literal/enum.rb

```diff
@@ -165,6 +165,11 @@ def deconstruct_keys(keys)
 	def _dump(level)
 		Marshal.dump(@value)
 	end
+
+	def ===(other)
+		return true if equal? other
+		@value == other
```

Comment:
I don't think this is correct. It would stop us from using case statements to check if a value actually is an enum member. This will return true even if the value is just the value that the enum tracks.

Reply:
Ok, but dont follow exactly where it breaks a membership check as thats done by `.===` no?

Reply:
No, it uses `===` on the object in the case statement. Here’s an example

```ruby
def message(color)
  case foo
  when Color::Blue
    "it’s blue"
  when Color::Red
    "it’s red"
  else
    raise "Invalid input"
  end
end

color = Color::Red
value = 1

message(color) # this should return "it’s red"
message(value) # this should raise
```

Reply:
Right ok of course!

Reply:
I guess I was wondering if an enum member instance could almost be interchangeable with its value in case statement, though pattern matching/deconstructing would solve any such use cases for that anyway so yeah agree doesn't make sense

Reply:
You can also just call .value

```ruby
case value
when Literal::Red.value
  # ...
```